### PR TITLE
Create fractional_cover_problem.py

### PR DESCRIPTION
**Pull request by debnath003**
### Describe your change:

Added Fractional Set Cover Problem Algorithm where you're allowed to take fractions of the elements to minimize the cost of covering a set of requirements. It is a variation of the well-known Knapsack Problem.

In the Fractional Cover Problem, you are given a set of items, each with a weight and a value, and a knapsack with a limited weight capacity. The goal is to determine how to select fractions of these items to maximize the total value placed in the knapsack without exceeding its weight capacity.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by debnath003** [comment]
@cclauss please approve this PR for hacktoberfest and add the hacktoberfest-accepted label after PR merging. Thank you!

File: greedy_methods/fractional_cover_problem.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Item:
+    weight: int
+    value: int
+
+    def __eq__(self, other: object) -> bool:
```
Comment (cclauss) [inline greedy_methods/fractional_cover_problem.py:11]: One of the really cool features of dataclasses is that this comes for free.  If you rename this function to `not_really__eq__()`, all your doctests still pass!!

File: greedy_methods/fractional_cover_problem.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Item:
+    weight: int
+    value: int
+
+    def not_really__eq__(self, other: object) -> bool:
```
[the remark this answers is not in the document]
Comment (debnath003) [reply greedy_methods/fractional_cover_problem.py:11]: Yeah I have done that and rechecked the doctest in my environment, it is running perfectly fine!

**Comment by cclauss** [comment]
Like this...

**Comment by debnath003** [comment]
> Like this...

Got this 😅 I was also trying to rename the __eq__()  to not_really__eq__() and got mypy right, but thanks for the suggestion! Now please approve this PR and give a hacktoberfest-accepted label for acceptance to hacktoberfest

**Comment by debnath003** [comment]
@cclauss please accept my PR and merge the PR for hacktoberfest, also give a hacktoberfest-accepted label for the acceptance! Awaiting your review.

**Comment by debnath003** [comment]
@cclauss please approve this PR for hacktoberfest, also give a hacktoberfest-accepted label for the acceptance! Awaiting your review.

**Comment by cclauss** [comment]
Please add test `>>> fractional_cover(items=[Item(10, 60)], capacity=-1)` which should raise a ValueError.

**Comment by debnath003** [comment]
> Please add test `>>> fractional_cover(items=[Item(10, 60)], capacity=-1)` which should raise a ValueError.

Done that @cclauss, please accept my PR and give a hacktoberfest-accepted label for the acceptance! Awaiting your review.

**Comment by cclauss** [comment]
Final edits: https://github.com/TheAlgorithms/Python/pull/9973/commits/b000f15c23ee47c0ab94294e13f0394bcc657dce

**Review by cclauss — APPROVED** [review]

**Comment by debnath003** [comment]
Please add hacktoberfest accepted label @cclauss

**Pull request merged**